cleanlab/rank.py

```diff
@@ -109,57 +123,335 @@ def get_self_confidence_for_each_label(
+      Set this = False if you prefer to emphasize label issues in classes that are more common
+      throughout the dataset
+      See paper "Confident Learning: Estimating Uncertainty in Dataset Labels" by Northcutt et al.
+      https://arxiv.org/abs/1911.00068
+
+    Returns
+    -------
+    label_quality_scores : np.array (float)
+      Scores are between 0 and 1 where lower scores indicate labels less likely to be correct.
+
+    See Also
+    --------
+    self_confidence
+    normalized_margin
+    confidence_weighted_entropy
+    subtract_confident_thresholds
+
+    """
+
+    # Available scoring functions to choose from
+    scoring_funcs = {
+        "self_confidence": get_self_confidence_for_each_label,
+        "normalized_margin": get_normalized_margin_for_each_label,
+        "confidence_weighted_entropy": get_confidence_weighted_entropy_for_each_label,
+    }
+
+    # Select scoring function
+    try:
+        scoring_func = scoring_funcs[method]
+    except KeyError:
+        raise ValueError(
+            f"""
+            {method} is not a valid scoring method for rank_by!
+            Please choose a valid rank_by: self_confidence, normalized_margin, confidence_weighted_entropy
+            """
+        )
+
+    # Adjust predicted probabilities
+    if adj_pred_probs:
+        pred_probs = subtract_confident_thresholds(labels, pred_probs)
+
+    # Pass keyword arguments for scoring function
+    input = {"labels": labels, "pred_probs": pred_probs}
+
+    # Calculate scores
+    label_quality_scores = scoring_func(**input)
+
+    return label_quality_scores
+
+
+def score_label_quality_ensemble(
+    labels: np.array,
+    pred_probs_list: List[np.array],
+    method: str = "self_confidence",
+    adj_pred_probs: bool = False,
+    weight_ensemble_members_by: str = "uniform",
+    verbose: int = 0,
+) -> np.array:
+    """Returns label quality scores based on predictions from an ensemble of models.
+
+    This is a function to compute label-quality scores for classification datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    Ensemble scoring requires a list of pred_probs from each model in the ensemble.
+
+    For each pred_probs in list, compute label quality score.
+    Take the average of the scores with the chosen weighting scheme determined by weight_ensemble_members_by.
+    Default weight_ensemble_members_by is "uniform" which computes the simple average of scores.
+
+    Score is between 0 and 1.
+    1 - clean label (given label is likely correct).
+    0 - dirty label (given label is likely incorrect).
+
+    Parameters
+    ----------
+    labels : np.array
+      Labels in the same format expected by the `score_label_quality()` method.
+
+    pred_probs_list : List of np.array (shape (N, K))
+      Each element in this list should be an array of pred_probs in the same format
+      expected by the `score_label_quality()` method
```

Comment:
Add extra sentence: 

Each element of `pred_probs_list` corresponds to the predictions from one model for all datapoints.

Reply:
Done, thanks